rtc.py

```diff
@@ -85,39 +106,20 @@ def system_call(params):
         print("\nERROR: Either " + args.granule + " does exist or it is not a GRD product.")
         exit(1)
 
-    print("\nDownloading Granule from " + download_url)
+    print("\nDownloading granule from " + download_url)
     write_netrc_file(args.username, args.password)
     local_file = download_file(download_url)
 
-    print("\nApplying Orbit File")
-    system_call(["gpt", "Apply-Orbit-File", "-Ssource=" + local_file, "-t",  "Orb"])
-    os.unlink(local_file)
-
-    print("\nRunning Calibration")
-    system_call(["gpt", "Calibration", "-PoutputBetaBand=true", "-PoutputSigmaBand=false", "-Ssource=Orb.dim", "-t", "Cal"])
-    delete_dim_files("Orb")
-
-    print("\nRunning Speckle Filter")
-    system_call(["gpt", "Speckle-Filter", "-Ssource=Cal.dim", "-t", "Spk"])
-    delete_dim_files("Cal")
-
-    print("\nRunning Terrain Flattening")
-    system_call(["gpt", "Terrain-Flattening", "-PreGridMethod=False", "-Ssource=Spk.dim", "-t", "TF"])
-    delete_dim_files("Spk")
-
-    print("\nRunning Terrain Correction")
-    system_call(["gpt", "Terrain-Correction", "-PpixelSpacingInMeter=30.0", "-PdemName=SRTM 1Sec HGT", "-Ssource=TF.dim", "-t", "TC"])
-
-    delete_dim_files("TF")
+    local_file = gpt(local_file, "Apply-Orbit-File")
```

Comment:
There's nothing wrong with how the following code is written and it will work. It's just the use of `local file` that bugs me. Personally, I would have written it more like: 

```
def gpt(input_file, command, *args):
    print("\n" + command)
    system_command = ["gpt", command, "-Ssource=" + input_file, "-t", command] + list(args)
    system_call(system_command)
    cleanup(input_file)
...
gpt("Apply-Orbit-File", local_file)
gpt("Calibration", "Apply-Orbit-File.dim", "-PoutputBetaBand=true", "-PoutputSigmaBand=false")
gpt("Speckle-Filter", "Calibration.dim")
gpt("Terrain-Flattening", "Speckle-Filter.dim",  "-PreGridMethod=False")
gpt("Terrain-Correction", "Terrain-Flattening.dim", "-PpixelSpacingInMeter=30.0", "-PdemName=SRTM 1Sec HGT")

for file_name in os.listdir("Terrain-Correction.data"):
   ...
```